src/api/app/controllers/trigger_controller.rb

```diff
@@ -1,53 +1,85 @@
 class TriggerController < ApplicationController
+  validate_action rebuild: { method: :post, response: :status }
+  validate_action release: { method: :post, response: :status }
   validate_action runservice: { method: :post, response: :status }
 
+  before_action :require_project_param, only: [:release]
+
+  before_action :extract_auth_from_request, :validate_auth_token, :require_valid_token
   #
-  # This controller is checking permission always only on the base of tokens
+  # Authentication happens with tokens, so no login is required
   #
   skip_before_action :extract_user
   skip_before_action :require_login
 
-  # github.com sends a hash payload
-  skip_before_action :validate_params, only: [:runservice]
-
-  def runservice
-    auth = request.env['HTTP_AUTHORIZATION']
+  # to get access to the method release_package
+  include MaintenanceHelper
 
-    if request.env['HTTP_X_GITLAB_EVENT'] == 'Push Hook'
-      auth = 'Token ' + request.env['HTTP_X_GITLAB_TOKEN']
-    end
+  include Trigger::Errors
 
-    unless auth && auth[0..4] == 'Token' && auth[6..-1] =~ /^[A-Za-z0-9+\/]+$/
-      render_error errorcode: 'permission_denied',
-                   message: "No valid token found 'Authorization' header",
-                   status: 403
-      return
-    end
+  def rebuild
+    Backend::Api::Sources::Package.rebuild(@pkg.project.name, @pkg.name)
+    render_ok
+  end
 
-    token = Token::Service.find_by_string(auth[6..-1])
+  def release
+    matched_repo = false
+    @pkg.project.repositories.includes(:release_targets).each do |repo|
+      repo.release_targets.where(trigger: 'manual').each do |releasetarget|
+        release_target_repository_project = releasetarget.repository.project
```

Comment:
This should be `releasetarget.target_repository.project`. releasetarget.repository would be the `repo` we are coming from

Reply:
could please take a look again @bgeuken ?